tests/integration/widgets/test_autocomplete_input.py

```diff
@@ -516,3 +516,29 @@ def test_server_on_change_round_trip_menu_entry(self, bokeh_server_page: BokehSe
         assert results['data']['val'] == ["400", "12344557"]
 
         assert page.has_no_console_errors()
+
+    def test_includes_search_strategy(self, bokeh_model_page: BokehModelPage) -> None:
+        # case_sensitive=True by default
+        text_input = AutocompleteInput(title="title", completions = ["123", "1123", "231", "1324", "3211"], search_strategy="includes")
+
+        page = bokeh_model_page(text_input)
+
+        # double click to highlight and overwrite old text
+        el = find_element_for(page.driver, text_input, "input")
+        enter_text_in_element(page.driver, el, "23", click=2, enter=False)
+
+        items = el.find_elements(By.TAG_NAME, "div")
+        assert len(items) == 3
+        assert items[0].text == "123"
+        assert items[1].text == "1123"
+        assert items[2].text == "231"
+        assert "bk-active" in items[0].get_attribute('class')
+
+        el = find_element_for(page.driver, text_input, "input")
+        enter_text_in_element(page.driver, el, "32", click=2, enter=False)
+
+        items = el.find_elements(By.TAG_NAME, "div")
+        assert len(items) == 2
+        assert items[0].text == "1324"
+        assert items[1].text == "3211"
+        assert "bk-active" in items[0].get_attribute('class')
```

Comment:
@minimav curious, did you actually run this test locally? These selenium tests are currently turned off in CI, because they became too flaky, currently we rely on browse tests driven from the BokehJS side rather than the Python side.

Reply:
Haha, good question - I thought I did but checking now I think I got confused about exactly which tests I had run on the Python side.

I just managed to get them running and there was one flaky test (`Test_AutocompleteInput.test_server_no_restriction`) and the one I added which was always failing. After a bit of fiddling I realised I was querying the wrong element in my new test and can now get them to all pass, albeit with the flaky one still 50-50 whether it passes.

Good spot :) 

![Screenshot 2023-03-29 at 20 03 09](https://user-images.githubusercontent.com/6566948/228641493-06ddf0d4-2216-4ce8-ba84-09b0d209a37e.png)